charts/redpanda/templates/post-install-upgrade-job.yaml

```
@@ -146,13 +149,13 @@ spec:
               path: tls.key
             - key: tls.crt
               path: tls.crt
-          {{- if $cert.caEnabled }}
+      {{- if $cert.caEnabled }}
             - key: ca.crt
               path: ca.crt
-          {{- end }}
-            secretName: {{ template "redpanda.fullname" $ }}-{{ $name }}-cert
-        {{- end }}
-      {{- end -}}
+      {{- end }}
+            secretName: {{ template "cert-secret-name" $r }}
```

Why the $cert can not be send to the template like
```suggestion
            secretName: {{ template "cert-secret-name" $cert }}
```

The template would need to change slightly.

---

The named template looks to see if this iteration's $cert has a secretRef. He adds a key to a copy of the root context (so as not to pollute the root) and passes the combination to "cert-secret-name". If secretRef is empty, that template includes a call to the "redpanda.fullname" template which needs the root context.

https://github.com/redpanda-data/helm-charts/pull/440/files#diff-8407f32f870c88088a3041f8a5a8394806973dabdef154b7fde83d5e77f70ed8R624-R630